pkg/types/defaults/installconfig.go

```diff
@@ -92,6 +95,11 @@ func SetInstallConfigDefaults(c *types.InstallConfig) {
 		}
 	}
 
+	if len(c.FeatureSet) == 0 && len(c.FeatureGates) > 0 {
+		c.FeatureSet = configv1.CustomNoUpgrade
+		logrus.Warn("Feature gates were provided in the Install Config without a feature set: setting FeatureSet to CustomNoUpgrade")
+	}
+
```

Comment:
Being implicit like this is generally against API guidelines. I think I made this an error in the past, if I didn't, this should be an error rather than an implicit decision here

Reply:
This makes sense. Will drop this commit.